packages/wallet-adapter/base/src/adapter.ts

```diff
@@ -143,7 +143,9 @@ export class StandardWalletAdapter extends BaseWalletAdapter implements Standard
 
             if (!this.#wallet.accounts.length) {
                 try {
-                    await this.#wallet.features[StandardConnect].connect(input);
+                    // `onlyIfTrusted` plays the same role as `silent`, some wallets like Phantom use `onlyIfTrusted` instead of `silent`.
+                    const connectInput = input?.silent ? { ...input, onlyIfTrusted: true } : input;
+                    await this.#wallet.features[StandardConnect].connect(connectInput);
```

Comment:
The purpose of calling this with `silent` is to imply intent using the Wallet Standard. The actual implementation is wallet-specific. `this.#wallet` is (e.g.) Phantom's Wallet Standard implementation, which should handle the `silent` option coming from the Standard and interpret it to use `onlyIfTrusted`.